dora/core/common/src/main/java/alluxio/wire/WorkerNetAddress.java

```diff
@@ -254,15 +254,38 @@ public boolean equals(Object o) {
         && mRpcPort == that.mRpcPort
         && mDataPort == that.mDataPort
         && mWebPort == that.mWebPort
-        && mDomainSocketPath.equals(that.mDomainSocketPath);
-        // Skip the comparison of mHttpServerPort for backward compatibility
+        && mDomainSocketPath.equals(that.mDomainSocketPath)
+        && mHttpServerPort == that.mHttpServerPort;
   }
 
   @Override
   public int hashCode() {
-    // Skip the mHttpServerPort for backward compatibility
     return Objects.hashCode(mSecureRpcPort, mHost, mContainerHost, mDataPort, mRpcPort, mWebPort,
-        mDomainSocketPath);
+        mDomainSocketPath, mHttpServerPort);
+  }
+
+  /**
+   * A customized equality comparison which skips the comparison of optional fields, including
+   * mHttpServerPort, for backward compatibility.
+   *
+   * @param o The object to be compared with this WorkerNetAddress for customized equality
+   * @return true if the specified object is equal to this WorkerNetAddress; false otherwise
+   */
+  public boolean customizedEquals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof WorkerNetAddress)) {
+      return false;
+    }
+    WorkerNetAddress that = (WorkerNetAddress) o;
+    return mHost.equals(that.mHost)
+        && mContainerHost.equals(that.mContainerHost)
+        && mSecureRpcPort == that.mSecureRpcPort
+        && mRpcPort == that.mRpcPort
+        && mDataPort == that.mDataPort
+        && mWebPort == that.mWebPort
+        && mDomainSocketPath.equals(that.mDomainSocketPath);
```

Comment:
I think this does not have to be put in `WorkerNetAddress` because the costumized comparison can be different depending on the caller. You can remove this and move code to `WorkerServiceEntity`.

Reply:
Sure. I moved the comparison logic to `WorkerServiceEntity`.